File: typescript-connector/src/models/AsPlannedLocalTransformation.ts
// Class from DFC Business Ontology: #AsPlannedLocalTransformation
import { SemanticObject } from "../core/SemanticObject.js";
import { Transformation, type TransformationParams } from "./Transformation.js";

export interface AsPlannedLocalTransformationParams extends TransformationParams {
  cost?: number;
  endDate?: string;
  startDate?: string;
  hasInput?: string;
  hasOutput?: string;
  transformedBy?: string;
}
export class AsPlannedLocalTransformation extends Transformation {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:AsPlannedLocalTransformation";
  }

  cost?: number;
  endDate?: string;
  startDate?: string;
  hasInput?: string;
  hasOutput?: string;
  transformedBy?: string;

  constructor(
    semanticId: string,
    params?: AsPlannedLocalTransformationParams,
  ) {
    super(semanticId, params);
    this.cost = params?.cost;
    this.endDate = params?.endDate;
    this.startDate = params?.startDate;
    this.hasInput = params?.hasInput;
    this.hasOutput = params?.hasOutput;
    this.transformedBy = params?.transformedBy;
    this.semanticType = AsPlannedLocalTransformation.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:cost", () => this.cost);
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:end_date", () => this.endDate);
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:start_date", () => this.startDate);
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:has_input", () => this.hasInput);
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:has_output", () => this.hasOutput);
    this.registerSemanticProperty("dfc-b:AsPlannedLocalTransformation:transformed_by", () => this.transformedBy);
  }
  static {
    SemanticObject.typeRegistry.set(AsPlannedLocalTransformation.SEMANTIC_TYPE, AsPlannedLocalTransformation);
  }
}

File: ruby-gem/lib/models/as_planned_local_transformation.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #AsPlannedLocalTransformation
require_relative '../core/semantic_object'
require_relative 'transformation'


module DfcLinkmlConnector
  module Models
    class AsPlannedLocalTransformation < Transformation
      SEMANTIC_TYPE = "dfc-b:AsPlannedLocalTransformation".freeze

      # @return [Float]
      attr_accessor :cost

      # @return [String]
      attr_accessor :end_date

      # @return [String]
      attr_accessor :start_date

      # @return [String]
      attr_accessor :input

      # @return [String]
      attr_accessor :output

      # @return [String]
      attr_accessor :transformed_by

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, cost: nil, endDate: nil, startDate: nil, input: nil, output: nil, transformedBy: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, cost: nil, endDate: nil, startDate: nil, input: nil, output: nil, transformedBy: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @cost = cost
        @end_date = endDate
        @start_date = startDate
        @input = input
        @output = output
        @transformed_by = transformedBy
        self.semanticType = "dfc-b:AsPlannedLocalTransformation"
        registerSemanticProperty("dfc-b:cost", &method("cost")).valueSetter = method("cost=")
        registerSemanticProperty("dfc-b:endDate", &method("end_date")).valueSetter = method("end_date=")
        registerSemanticProperty("dfc-b:startDate", &method("start_date")).valueSetter = method("start_date=")
        registerSemanticProperty("dfc-b:hasInput", &method("input")).valueSetter = method("input=")
        registerSemanticProperty("dfc-b:hasOutput", &method("output")).valueSetter = method("output=")
        registerSemanticProperty("dfc-b:transformedBy", &method("transformed_by")).valueSetter = method("transformed_by=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
